src/realm/util/file.cpp

```diff
@@ -835,12 +835,6 @@ void File::prealloc(size_t size)
 #endif
     };
 
-#if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
```

Comment:
Does musl not define these POSIX version cpp defines? What exactly is this change here doing?

Reply:
Yeah, musl doesn't define _POSIX_C_SOURCE, because it's meant to be used in a different way than how realm uses it - see <https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_02_01>. Basically, it's intended to be used by programs to ask an implementation which is known to support a given version of POSIX to expose the definitions associated with that version, not to test whether or not the implementation supports that version of POSIX. As I mentioned above, I'm not sure what the right way to test this is, so I opted to just make the path without _POSIX_C_SOURCE functional. The follow-up here would be changing prealloc_if_supported and is_prealloc_supported to use some other yet-to-be-determined method to figure out whether or not posix_fallocate is supported.